doc/lms.mysql

```diff
@@ -3302,13 +3346,309 @@ CREATE VIEW vnodes AS
+            ELSE a.id
+        END) AS suspension_assignment_id,
+        (CASE WHEN suspensions.suspension_id IS NOT NULL OR suspensions_all.suspend_all = 1 THEN 1 ELSE 0 END) AS suspended,
+        (CASE WHEN suspensions.suspension_id IS NULL AND suspensions_all.suspend_all = 1 THEN 1 ELSE 0 END) AS suspension_suspend_all,
+        (CASE WHEN suspensions.suspension_id IS NOT NULL
+            THEN suspensions.suspension_id
+            ELSE suspensions_all.suspension_id
+        END) AS suspension_id,
+        (CASE WHEN suspensions.suspension_id IS NOT NULL
+            THEN suspensions.at
+            ELSE suspensions_all.at
+        END) AS suspension_at,
+        (CASE WHEN suspensions.suspension_id IS NOT NULL
+            THEN suspensions.datefrom
+            ELSE suspensions_all.datefrom
+        END) AS suspension_datefrom,
+        (CASE WHEN suspensions.suspension_id IS NOT NULL
+            THEN suspensions.dateto
+            ELSE suspensions_all.dateto
+        END) AS suspension_dateto,
+        (CASE WHEN suspensions.suspension_id IS NOT NULL
+            THEN suspensions.chargemethod
+            ELSE suspensions_all.chargemethod
+        END) AS suspension_charge_method,
+        (CASE WHEN suspensions.suspension_id IS NOT NULL
+            THEN suspensions.calculationmethod
+            ELSE suspensions_all.calculationmethod
+        END) AS suspension_calculation_method,
+        (CASE WHEN suspensions.suspension_id IS NOT NULL
+            THEN suspensions.value
+            ELSE suspensions_all.value
+        END) AS suspension_value,
+        (CASE WHEN suspensions.suspension_id IS NOT NULL
+            THEN suspensions.percentage
+            ELSE suspensions_all.percentage
+        END) AS suspension_percentage,
+        (CASE WHEN suspensions.suspension_id IS NOT NULL
+            THEN suspensions.netflag
+            ELSE suspensions_all.netflag
+        END) AS suspension_netflag,
+        (CASE WHEN suspensions.suspension_id IS NOT NULL
+            THEN suspensions.currency
+            ELSE suspensions_all.currency
+        END) AS suspension_currency,
+        (CASE WHEN suspensions.suspension_id IS NOT NULL
+            THEN suspensions.taxid
+            ELSE suspensions_all.taxid
+        END) AS suspension_tax_id,
+        (CASE WHEN suspensions.suspension_id IS NOT NULL
+            THEN suspensions.note
+            ELSE suspensions_all.note
+        END) AS suspension_note,
+        (CASE WHEN suspensions.customerid IS NOT NULL
+            THEN suspensions_all.customerid
+            ELSE a.customerid
+        END) AS suspension_customer_id,
+        (CASE WHEN suspensions.taxrate IS NOT NULL
+            THEN suspensions.taxrate
+            ELSE suspensions_all.taxrate
+        END) AS suspension_taxrate,
+        (CASE WHEN suspensions.taxlabel IS NOT NULL
+            THEN suspensions.taxlabel
+            ELSE suspensions_all.taxlabel
+        END) AS suspension_taxlabel,
```

Comment:
W takich miejsach nie móżna się posłużyć w sprytny sposób funkcją SQL `COALESCE()`?

Reply:
Tu akurat powinno być:
`(CASE WHEN **suspensions.suspension_id** IS NOT NULL
            THEN suspensions.taxlabel
            ELSE suspensions_all.taxlabel
        END) AS suspension_taxlabel,`
tak jak w innych polach tego zapytania co jest moim przeoczeniem i to poprawiłem. W takim przypadku COALECSE() jest nieuzasadnione. To co było wcześniej działało, ale mogło niespodziewanie się zachować w pewnych nietypowych sytuacjach.  

Wprowadziłem COALECSE() dla innego pola w tym zapytaniu, bo się dało.